symplyphysics/core/geometry/line.py

```diff
@@ -0,0 +1,20 @@
+from sympy import symbols, Point, Line, Expr, solve
+
+def two_point_function(p1: Point, p2: Point, x: Expr) -> Expr:
+    """Constructs a linear function of ``x`` using two points ``p1`` and ``p2``.
+    
+    Returns ``NotImplemented`` if the line equation does not depend on ``y``.
+    """
+
+    xsym = x
+
+    x, y = symbols("x y")
+    line = Line(p1, p2)
+    equation = line.equation(x=x, y=y)  # type: ignore[attr-defined]
+    
+    y_functions = solve(equation, y)
+
+    if not y_functions:
+        return NotImplemented
+
```

Comment:
Also raise on the same Point instead of 2 different points.

Reply:
`Line2D` would have raised it already, so no need.

Reply:
So we need a test to show it.